plugin/plaintextmodel.cpp

```diff
@@ -117,6 +121,11 @@ void PlainTextModel::setSource(const QUrl &source)
                 m_status = Error;
             } else {
                 m_textStream.setDevice(&m_file);
+                QByteArray ba = m_file.readAll();
+
+                QString name = detect(ba.data(), ba.size())->name();
+                m_textStream.setCodec(QTextCodec::codecForName( name.toStdString().data() ));
```

Comment:
Uhm, so the method returns qtextcodec pointer, which is converted to string which is converted to qbytearray which is converted back to different qtextcodec pointer?

Reply:
Yes, It's a mistake.